### PR TITLE
Adding location support

### PR DESCRIPTION
**Pull request by ianisms**
#256 
- Added support for getting locations
- Added support for getting / setting location mode

### CODE REVIEW

File: ring_doorbell/location.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        self._latitude = locationData["geo_coordinates"]["latitude"]
+        self._longitude = locationData["geo_coordinates"]["longitude"]
+        self._address1 = locationData["address"]["address1"]
+        self._address2 = locationData["address"]["address2"]
+        self._city = locationData["address"]["city"]
+        self._state = locationData["address"]["state"]
+        self._zip_code = locationData["address"]["zip_code"]
+        self._country = locationData["address"]["country"]
+        self._timezone = locationData["address"]["timezone"]
+        self.update()
```
Comment (balloob) [inline ring_doorbell/location.py:36]: We shouldn’t query in constructor. It should just init the class, no side effects. Also, you just got data?

File: ring_doorbell/location.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    def country(self):
+        """Return location country."""
+        return self._country
+
+    @property
+    def timezone(self):
+        """Return location timezone."""
+        return self._timezone
+
+    @mode.setter
```
Comment (balloob) [inline ring_doorbell/location.py:115]: We shouldn’t use setters. Use a method and have the parameters be the same as what the endpoint accepts.

File: ring_doorbell/location.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        """Return location address1."""
+        return self._address1
+
+    @property
+    def address2(self):
+        """Return location address2."""
+        return self._address2
+
+    @property
+    def city(self):
```
Comment (balloob) [inline ring_doorbell/location.py:91]: All these properties are meh, let’s just assign things to `self.address2` etc in the constructor.

**Comment by firefart** [comment]
any update on this? It would be awesome to get the current location status so we can have some hass automations based on this. Happy to update the PR if you want to adress the comments

**Pull request closed without merging**